connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -419,8 +420,14 @@ public ConnectorMetricsGroup(ConnectMetrics connectMetrics, AbstractStatus.State
             metricGroup.close();
 
             metricGroup.addImmutableValueMetric(registry.connectorType, connectorType());
-            metricGroup.addImmutableValueMetric(registry.connectorClass, connector.getClass().getName());
-            metricGroup.addImmutableValueMetric(registry.connectorVersion, connector.version());
+            metricGroup.addImmutableValueMetric(registry.connectorClass, connector.pluginClass().getName());
+            String version;
+            try {
+                version = connector.version();
+            } catch (Exception e) {
+                version = DelegatingClassLoader.UNDEFINED_VERSION;
+            }
```

Comment:
This technically changes behavior, right? We go from failing the connector on startup to just assigning it an undefined version.

I think this is probably fine (the benefits outweigh the costs, and I have a hard time imagining any use case that would favor failing here), just want to make sure I'm gauging the impact here correctly.